src/libaktualizr/primary/eventsinterpreter.cc

```diff
@@ -29,11 +29,20 @@ std::shared_ptr<command::BaseCommand> EventsInterpreter::handle_cycle(event::Bas
   } else if (event.variant == "FetchMetaComplete") {
     return std::make_shared<command::CheckUpdates>();
   } else if (event.variant == "UpdateAvailable") {
-    return std::make_shared<command::StartDownload>(dynamic_cast<event::UpdateAvailable &>(event).updates);
+    auto update_available = dynamic_cast<event::UpdateAvailable &>(event);
+    pending_ecus = update_available.ecus_count;
```

Comment:
Counting the number of ecus on this side seems brittle.

For example if [this condition](https://github.com/advancedtelematic/aktualizr/blob/73176f224f1dc13c88151c2e06d51dfdda2ffbb3/src/libaktualizr/primary/sotauptaneclient.cc#L1023) is taken, one `InstallationComplete` event will be missing and the machine will never go to the next state. Note: it would be good to have a test for this.

We could consider having to different events, one for each ecu complete installation and another for the complete system update. The decision of whether the system is updated completely will be computed in `sotauptaneclient.cc` and `eventsinterpreter.cc` will stay simple and mechanical.